src/main/java/org/benetech/servicenet/adapter/healthleads/JsonDataResolver.java

```diff
@@ -0,0 +1,63 @@
+package org.benetech.servicenet.adapter.healthleads;
+
+import com.google.gson.Gson;
+import org.benetech.servicenet.adapter.healthleads.model.*;
+import org.benetech.servicenet.type.ListType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class JsonDataResolver {
+
+    public List<BaseData> getDataFromJson(final String json, final String filename) {
+        DataType type;
+        try {
+            type = DataType.valueOf(filename.toUpperCase());
+        } catch (IllegalArgumentException | NullPointerException e) {
+            return new ArrayList<>();
+        }
+        switch (type) {
+            case ELIGIBILITY:
+                return getDataFromJson(json, Eligibility.class);
+            case LANGUAGES:
+                return getDataFromJson(json, Language.class);
+            case LOCATIONS:
+                return getDataFromJson(json, Location.class);
+            case ORGANIZATIONS:
+                return getDataFromJson(json, Organization.class);
+            case PHONES:
+                return getDataFromJson(json, Phone.class);
+            case PHYSICAL_ADDRESSES:
+                return getDataFromJson(json, PhysicalAddress.class);
+            case REQUIRED_DOCUMENTS:
+                return getDataFromJson(json, RequiredDocument.class);
+            case SERVICES:
+                return getDataFromJson(json, Service.class);
+            case SERVICES_AT_LOCATION:
+                return getDataFromJson(json, ServiceAtLocation.class);
+            case SERVICES_TAXONOMY:
+                return getDataFromJson(json, ServiceTaxonomy.class);
+            case TAXONOMY:
+                return getDataFromJson(json, Taxonomy.class);
+        }
+        return new ArrayList<>();
+    }
+
+    private List<BaseData> getDataFromJson(final String json, Class<? extends BaseData> clazz) {
+        return new Gson().fromJson(json, new ListType<>(clazz));
+    }
+
+    private enum DataType {
```

Comment:
I think that getting classes by name is not a good idea. Some files are named in plural form and if we would want to get classes by this name we would have to name classes in plural form also or provide some logic to resolve filenames to a singular form.

Reply:
You are currently getting enum values by name, so adding fields to the enum values, like:
LOCATION(Location.class),
SERVICE(Service.class)
would simplify the code above to:
type.getClazz() instead of the big switch statement